workflow/scripts/osemosys_global/powerplant/residual_capacity.py

```diff
@@ -0,0 +1,124 @@
+"""Function to calculate residual capacity for powerplant technologies."""
+
+import pandas as pd
+import itertools
+
+from data import(
+    create_pwr_techs, 
+    get_years
+    )
+
+def res_capacity(df_gen_base, tech_list, df_tech_code, duplicate_techs, 
+                 start_year, end_year, region_name):
```

Comment:
Similar to other comments. `tech_list` and `df_tech_code` are not used here?

Reply:
Correct!